README.md

```diff
@@ -1,2 +1,188 @@
+NAME:
+   spacelift-cli stack deploy - Start a deployment (tracked run)
+
+USAGE:
+   spacelift-cli stack deploy [command options] [arguments...]
+
+CATEGORY:
+   Run management
+
+OPTIONS:
+   --sha value  Commit SHA for the newly created run
+   --tail       Indicate whether to tail the run (default: false)
+   --help, -h   show help (default: false)
+```
+
+This subcommand creates a [tracked run](https://docs.spacelift.io/concepts/run/tracked) on a stack. By default, it uses the latest tracked commit, but you can also point it to an arbitrary SHA using the `--sha` flag. Also by default, this command only creates a run and prints out the URL where the run can be accessed. If the `--tail` flag is set however, run logs will be retrieved until the run terminates. If the run is tailed, the exit code of the command will depend on the outcome of the run - if the run is `FINISHED`, the command succeeds (exits with `0`), otherwise it fails.
+
+Notes:
+
+- terminating the executable (Ctrl+C) while tailing logs does not stop the run, it just stops tailing logs;
+
+- you cannot confirm or discard the run from the command line. If the deployment ends up in [`UNCONFIRMED`](https://docs.spacelift.io/concepts/run/tracked#unconfirmed) state, the CLI be stuck until the state machine progresses. You will need to go to the run URL and make a decision there;
+
+#### `stack preview` subcommand
+
+```bash
+❯ spacelift-cli stack --id=stack-id preview --help
+NAME:
+   spacelift-cli stack preview - Start a preview (proposed run)
+
+USAGE:
+   spacelift-cli stack preview [command options] [arguments...]
+
+CATEGORY:
+   Run management
+
+OPTIONS:
+   --sha value  Commit SHA for the newly created run
+   --tail       Indicate whether to tail the run (default: false)
+   --help, -h   show help (default: false)
+```
+
+This subcommand creates a [proposed run](https://docs.spacelift.io/concepts/run/proposed) on a stack. By default, it uses the latest tracked commit, but you can also point it to an arbitrary SHA using the `--sha` flag. Also by default, this command only creates a run and prints out the URL where the run can be accessed. If the `--tail` flag is set however, run logs will be retrieved until the run terminates. If the run is tailed, the exit code of the command will depend on the outcome of the run - if the run is `FINISHED`, the command succeeds (exits with `0`), otherwise it fails.
+
+#### `stack logs` subcommand
+
+```bash
+❯ spacelift-cli stack --id=stack-id logs --help
+NAME:
+   spacelift-cli stack logs - Show logs for a particular run
+
+USAGE:
+   spacelift-cli stack logs [command options] [arguments...]
+
+CATEGORY:
+   Run management
+
+OPTIONS:
+   --run value  ID of the run
+   --help, -h   show help (default: false)
+```
+
+This subcommand prints out (and tails) logs for an existing run or task. The exit code of the command will depend on the outcome of the run - if the run is `FINISHED`, the command succeeds (exits with `0`), otherwise it fails.
```

Comment:
This feels a little weird in the sense that the `logs` command may have succeeded, but we'll return a non-zero exit code depending on the state of the run. I wonder if a separate command to get the run state would make more sense?

Did you have any use-case in mind for this, or was it just arbitrary?

Reply:
Code reuse :) I guess you're right, this sucks.

One way around it would be for the logs subroutine to return a tuple of `(*RunStateTransition, error)` and have the caller decide what to do about it. I'll try to fix it in this iteration.

Reply:
Done.